src/slurm_plugin/slurm_resources.py

```diff
@@ -414,7 +420,12 @@ def is_backing_instance_valid(self, log_warn_if_unhealthy=True):
                         self,
                         self.state_string,
                     )
-                return False
+                # Allow a few minutes for the eventual consistency of EC2 data
+                # This can only be done if the start time and timeout exist
+                if self.slurmdstarttime is None or ec2_backing_instance_timeout is None:
+                    return False
+                if (datetime.now() - self.slurmdstarttime).seconds > ec2_backing_instance_timeout:
+                    return False
```

Comment:
* I think datetime.now() is not enough, because you should consider the timezone, i.e. datetime.now(tz=timezone.utc)
* instead of doing this double if, you should leverage the `time_is_up` util function we have in the common package
* you should add logging messaging to understand when what's happening here, e.g. instance is not present but  it's within the timeout Vs instance is not present and timeout expired
* unit tests are missing for this new logic

Reply:
Added unit tests and switched to using the count interval since the start time is unknown

Reply:
Added additional logging as well